packages/slice-machine/lib/models/common/widgets/StructuredText/Form.tsx

```diff
@@ -0,0 +1,222 @@
+import { DefaultFields } from "@lib/forms/defaults";
+import options, { optionValues } from "./options";
+import WidgetFormField from "@lib/builders/common/EditModal/Field";
+import { Text, Button, Label, Checkbox, Flex, Box } from "theme-ui";
+import { Col, Flex as FlexGrid } from "@components/Flex";
+import SliceMachineIconButton from "@components/SliceMachineIconButton";
+
+import { has } from "fp-ts/Record";
+import { AnyObject } from "yup/lib/types";
+import { isObject, isString } from "lodash";
+
+const isAllSet = (curr: Array<string>): boolean =>
+  !optionValues.find((e) => !curr.includes(e));
+
+const _createInitialOptions = (str: string) => {
+  const arr = str.split(",");
+  return options.reduce<Array<string>>((acc, { value }) => {
+    if (arr.includes(value)) {
+      return [...acc, value];
+    }
+    return acc;
+  }, []);
+};
+
+function getValueFromYupContext(
+  ctx: yup.TestContext<AnyObject>,
+  key: string
+): string | null {
+  if (
+    has("parent", ctx) &&
+    isObject(ctx.parent) &&
+    has(key, ctx.parent) &&
+    isString(ctx.parent[key])
+  )
+    return ctx.parent[key];
+  return null;
+}
+
+function createValidator(key: "single" | "multi") {
+  return () =>
+    yup.string().test({
+      name: key,
+      message: `Options cannot be empty`,
+      test: function (_, ctx) {
+        const other = key === "single" ? "multi" : "single";
+        const val = getValueFromYupContext(ctx, key);
+        const otherVal = getValueFromYupContext(ctx, other);
+
+        if (otherVal && typeof otherVal === "string") {
+          return true;
+        }
+        if (typeof val !== "string") {
+          return false;
+        }
+        if ((!val || !val.length) && (!otherVal || !otherVal.length)) {
+          return false;
+        }
+        const arr = val.split(",");
+        if (arr.find((e) => !optionValues.includes(e))) {
+          return false;
+        }
+        return true;
+      },
+    });
+}
+
+const FormFields = {
+  ...DefaultFields,
+  allowTargetBlank: CheckBoxConstructor("Allow target blank for links"),
+  single: {
+    validate: createValidator("single"),
+  },
+  multi: {
+    validate: createValidator("multi"),
+  },
+};
+
+const accessors = ["config.single", "config.multi"];
+
+type Config = {
+  allowTargetBlank: boolean;
+  label: string; // RichText
+  placeholder: string;
+  single?: string;
+  multi?: string;
+};
+
+type FormValues = {
+  config: Config;
+  id: string;
+};
+
+const WidgetForm: React.FC<{
+  initialValues: FormValues;
+  values: FormValues & { mockConfig: Record<string, string | number> };
+  errors: FormValues;
+  fields: Array<Record<string, unknown>>; // fields of all the widgets
+  setFieldValue: (a: string, b?: string) => void;
+}> = (props) => {
+  const {
+    initialValues,
+    values: formValues,
+    errors,
+    fields,
+    setFieldValue,
+  } = props;
+
+  const {
+    config: { single, multi },
```

Comment:
this field is super weird 😂 , how we register either single or the multi property.

Reply:
eeyup, that's what the old file did so there's a few weird parts,

Reply:
both get registered